### PR TITLE
[release-4.13] OCPBUGS-29721: Add existing kubeletconfig/ctrcfg mc-name-suffix annotation

### PR DESCRIPTION
**Pull request by QiWang19**
**- What I did**
cherry-pick of https://github.com/openshift/machine-config-operator/pull/4187
replace https://github.com/openshift/machine-config-operator/pull/4205
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by QiWang19** [comment]
@yuqi-zhang PTAL

**Review by yuqi-zhang — APPROVED** [review]
/lgtm

What was the issue with the automatic backport?

File: pkg/controller/container-runtime-config/helpers.go [hunk 1]
```diff
@@ -253,6 +303,15 @@ func getManagedKeyCtrCfg(pool *mcfgv1.MachineConfigPool, client mcfgclientset.In
 	return fmt.Sprintf("99-%s-generated-containerruntime-%s", pool.Name, strconv.Itoa(suffixNum+1)), nil
 }
 
+func notLatestContainerRuntimeConfigInPool(ctrcfgList []mcfgv1.ContainerRuntimeConfig, cfg *mcfgv1.ContainerRuntimeConfig) bool {
+	for _, crc := range ctrcfgList {
+		if cfg.CreationTimestamp.Time.Before(crc.CreationTimestamp.Time) {
```
Comment (QiWang19) [inline pkg/controller/container-runtime-config/helpers.go:308]: > What was the issue with the automatic backport?

@yuqi-zhang  The `Compare` not defined, I changed to use the `Before()`
```
 pkg/controller/kubelet-config/helpers.go:347:27: kc.CreationTimestamp.Compare undefined (type "k8s.io/apimachinery/pkg/apis/meta/v1".Time has no field or method Compare) (typecheck)
package kubeletconfig 
```

I think this is ok, covered by thectrcfg/kcfg config_controller_test.go

**Comment by sunilcio** [comment]
The first kubeletconfig got the annotation set to ""
```
% oc get clusterversion
NAME      VERSION                                                   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.13.0-0.ci.test-2024-03-06-091948-ci-ln-tjslvlk-latest   True        False         49s     Cluster version is 4.13.0-0.ci.test-2024-03-06-091948-ci-ln-tjslvlk-latest

% oc label mcp worker aro.openshift.io/limits=""
machineconfigpool.machineconfiguration.openshift.io/worker labeled

% oc apply -f ../debug1/kc-aro.yaml 
kubeletconfig.machineconfiguration.openshift.io/aro-limits created

% oc apply -f ../debug1/kc-dynamic.yaml 
kubeletconfig.machineconfiguration.openshift.io/dynamic-node created

% oc get kubeletconfig aro-limits -o yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  annotations:
    machineconfiguration.openshift.io/mc-name-suffix: ""
  creationTimestamp: "2024-03-06T09:51:34Z"
  finalizers:
  - 99-worker-generated-kubelet
  generation: 1
  labels:
    aro.openshift.io/limits: ""
  name: aro-limits
  resourceVersion: "30342"
  uid: de816a4b-9f68-46c1-a08a-ace7ad8535de
spec:
  kubeletConfig:
    evictionHard:
      imagefs.available: 15%
      memory.available: 500Mi
      nodefs.available: 10%
      nodefs.inodesFree: 5%
    systemReserved:
      memory: 2000Mi
  machineConfigPoolSelector:
    matchLabels:
      aro.openshift.io/limits: ""
status:
  conditions:
  - lastTransitionTime: "2024-03-06T09:51:34Z"
    message: Success
    status: "True"
    type: Success

% oc get kubeletconfig dynamic-node -o yaml
apiVersion: machineconfiguration.openshift.io/v1
kind: KubeletConfig
metadata:
  annotations:
    machineconfiguration.openshift.io/mc-name-suffix: "1"
  creationTimestamp: "2024-03-06T09:51:47Z"
  finalizers:
  - 99-worker-generated-kubelet-1
  generation: 1
  name: dynamic-node
  resourceVersion: "30490"
  uid: ee76ca24-3fbe-4fff-9bf6-a188756f246a
spec:
  autoSizingReserved: true
  machineConfigPoolSelector:
    matchLabels:
      aro.openshift.io/limits: ""
status:
  conditions:
  - lastTransitionTime: "2024-03-06T09:51:47Z"
    message: Success
    status: "True"
    type: Success

```

**Pull request merged**